tests/models/test_data/test_bad_structures.json

```diff
@@ -2371,37 +2377,5 @@
       }
     ],
     "structure_features": []
-  },
-  {
-    "task_id": "db/1234567",
-    "type": "structure",
-    "last_modified": {
-      "$date": "2019-06-08T05:13:37.331Z"
-    },
-    "band_gap": 1.23456,
-    "chemsys": "Ge-Si",
-    "elements": ["Ge", "Si"],
-    "nsites": 3,
-    "nelements": 2,
-    "elements_ratios": [0.5, 0.5],
-    "pretty_formula": "GeSi",
-    "formula_anonymous": "AB",
-    "dimension_types": [1, 1, 1],
-    "nperiodic_dimensions": null,
```

Comment:
This is also a valid document now

Reply:
That's fine, I guess. The check here for `nperiodic_dimensions` should rather be "if `dimension_types` is provided, and a value is provided for `nperiodic_dimensions`, then it MUST equal the sum of `1`s in `dimension_types`"...